### ISSUE TITLE
Make agent tracing require the forwarder to be running?

### ISSUE THREAD

**Issue by jodh-intel** [kata-containers/kata-containers#2363]
As part of testing https://github.com/kata-containers/kata-containers/pull/2353, I noticed that the trace forwarder no longer needs to be running if agent tracing is enabled. Previously, if you enabled agent tracing, the agent would die with an explanatory message immediately if it could not connect to the trace forwarder.

The change was made on https://github.com/kata-containers/kata-containers/pull/2235 (which I approved, but must have missed that part! ;)

In summary: do we want to revert to the old behaviour? It might be a bit brutal, but I think it's less surprising for the agent to die with an error message if the user requested tracing but the agent is unable to provide that function, but what do others think? Currently, unless you realise you have to **manually** start the trace forwarder, enabling agent tracing will be almost an NOP (you won't get any trace spans, but you will get errors in the logs about being unable to create trace spans).

Alternatively, if `agent.enable_tracing=true` is set, we _could_ get the runtime to spawn the trace forwarder. That would be helpful, but brings additional costs in terms of a runtime code bloat and the time to write more tests. However, since it's unclear what the long-term future of the trace forwarder is, I'm not sure this is the right answer.

/cc @Tim-Zhang, @egernst, @cmaf, @kata-containers/architecture-committee.

**Comment by Tim-Zhang** [comment]
@jodh-intel The change in my pr is intend to introduce the re-connect mechanism so that we can start the trace-forwarder anytime. This has some benefits:
1. The trace-forwarder could be launch after or before the agent running.
2. If the trace-forwarder is crushed, it could be restart and the tracing could keep going.

**Comment by c3d** [comment]
I personally prefer the last option you suggest. Is it that much code bloat? It would be transparent, and I'm not sure it's that much more code than emitting an error message.

**Comment by Tim-Zhang** [comment]
@c3d , I agree, set up trace-forwarder in the runtime is easy to use,  but there may be multi trace-forwarder on one host, could the overhead be acceptable?
